crates/net/network/src/peers/manager.rs

```diff
@@ -1008,9 +1025,10 @@ impl PeerBackoffDurations {
     }
 
     /// Returns the timestamp until which we should backoff
-    pub fn backoff_until(&self, kind: BackoffKind) -> std::time::Instant {
+    pub fn backoff_until(&self, kind: BackoffKind, backoff_counter: u32) -> std::time::Instant {
+        let backoff_time = self.backoff(kind) * (backoff_counter + 1);
```

Comment:
hmm, this +1 seems redundant if we also do +1 after we call the function.
it's also not obvious that the function would take the arg  `backoff_counter` and increment it.

this should be applied before calling the function, we can either start the counter at 1 and rename the var or increment before calling `backoff_until`.

Reply:
Makes sense, it will be more readable this way. I pushed the update